### PR TITLE
Handle no main media in Immersive layout

### PR DESCRIPTION
**Pull request by buck06191**
## What does this change?
Add handling of immersive articles that don't have main media

Co-authored with @oliverlloyd 

### Changes
- Add story and fixture for `PrintShop` article (an example use of `noMainMedia`.
- Add `noMainMedia` prop to `ArticleHeadline` and `ArticleTitle` components.
- Styling changes for the article headline where main media isn't present.

### Desktop

| Before  (at 50% zoom to fit)    | After      |
|-------------|------------|
| ![beforeDesktop][] | ![afterDesktop][] |

[beforeDesktop]: https://user-images.githubusercontent.com/9122944/104348868-1f61ea80-54fa-11eb-8b66-de694e756e90.png
[afterDesktop]: https://user-images.githubusercontent.com/9122944/104348990-41f40380-54fa-11eb-92ee-f2b1f017d4d3.png

### Mobile
| Before | After      |
|-------------|------------|
| ![beforeMobile][] | ![afterMobile][] |

[beforeMobile]: https://user-images.githubusercontent.com/9122944/104349258-9a2b0580-54fa-11eb-98ca-f5d2332ca4dc.png
[afterMobile]: https://user-images.githubusercontent.com/9122944/104349215-8c758000-54fa-11eb-89c6-3026c47def05.png

## Why?
Edge case for Immersive layouts. Needed to get parity for DCR.

### CODE REVIEW

**Comment by buck06191** [comment]
@oliverlloyd Let me know if I missed anything else major in the changes that you did in that first commit.

**Comment by buck06191** [comment]
[Print shop tagged pieces](https://trello.com/c/ebZLnnER/2248-print-shop-tagged-pieces)

File: .prettierignore [hunk 1]
```diff
@@ -14,3 +14,5 @@ node_modules/
 !/src/**/*.ts
 !/src/**/*.tsx
 !/src/**/*.js
+# The below causes a bunch of prettier warnings, but we might also want to be formatting these.
+# !/fixtures/**/*.ts
```
Comment (oliverlloyd) [inline .prettierignore:18]: Might be worth enabling this in a separate PR. It's always nice to have pretty things!

File: src/web/components/ArticleHeadlinePadding.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const ArticleHeadlinePadding: React.FC<{
 	children: React.ReactNode;
 	designType: DesignType;
 }> = ({ children, designType }) => {
-	const paddingClassName = determinPadding(designType);
+	const paddingClassName = determinePadding(designType);
```
Comment (oliverlloyd) [inline src/web/components/ArticleHeadlinePadding.tsx:43]: Merci

**Review by oliverlloyd — APPROVED** [review]
Approving this from my perspective but it'd be great to get someone else's eyes on this.

I'm not crazy about this `noMainMedia` prop. It's a lot of prop for a very small edge case. I think this was what made me pause earlier.

File: src/web/layouts/ImmersiveLayout.tsx [hunk 3]
```diff
@@ -273,12 +272,12 @@ export const ImmersiveLayout = ({
                         100vw is normally enough but don't let the content shrink vertically too
                         much just in case
                     */
-					min-height: 25rem;
+					min-height: ${mainMedia && '25rem'};
```
Comment (liywjl) [inline src/web/layouts/ImmersiveLayout.tsx:275]: if `mainMedia` is not defined does this set `min-height: undefined`? if so maybe worth not setting the key?

```suggestion
					${mainMedia && 'min-height: 25rem'};
```

**Review by liywjl — APPROVED** [review]
Really great work! 👍  just a couple points about remove CSS prop keys if condition is not met

**Comment by paperboyo** [comment]
Great to see that! 👏 
Worth pinging AR (not sure how support looks there)?

File: src/web/layouts/ImmersiveLayout.tsx (hunk 3 above)
Comment (buck06191) [reply src/web/layouts/ImmersiveLayout.tsx:275]: Looks like doing this throws an error inside the `css` template. It needs a colon after a string to register as correct

Comment (buck06191) [reply src/web/layouts/ImmersiveLayout.tsx:275]: Opting for
```javascript
const noMainMediaStyles = css`
	height: 100vh;
	/**
    100vw is normally enough but don't let the content shrink vertically too
    much just in case
    */
	min-height: 25rem;
	${from.desktop} {
		min-height: 31.25rem;
	}
	${from.wide} {
		min-height: 50rem;
	}
`;
```
and then using 
```javascript
className={cx(
    !mainMedia && noMainMediaStyles,
    css`
        display: flex;
	flex-direction: column;
    `,
)}
```

File: .prettierignore (hunk 1 above)
Comment (buck06191) [reply .prettierignore:18]: Aye, I've kept it here partly because I didn't want to have to fight prettier again if the fixture needed updating. Once this is merged I'll do a PR to uncomment it and prettify everything

Comment (buck06191) [reply .prettierignore:18]: And also because otherwise I feel like I'd have to squash the commit history down to explain away why it was added and then removed again.

File: src/web/layouts/ImmersiveLayout.tsx (hunk 3 above)
Comment (liywjl) [reply src/web/layouts/ImmersiveLayout.tsx:275]: yeah that looks a lot nicer! 👍
